qutebrowser/browser/commands.py

```diff
@@ -620,30 +620,35 @@ def scroll_page(self, x: float, y: float, *,
                 "representation.")
 
     @cmdutils.register(instance='command-dispatcher', scope='window')
-    def yank(self, title=False, sel=False, domain=False, pretty=False):
-        """Yank the current URL/title to the clipboard or primary selection.
+    @cmdutils.argument('what', choices=['selection', 'url', 'pretty-url',
+                                        'title', 'domain'])
+    def yank(self, what='url', sel=False, keep=False):
+        """Yank something to the clipboard or primary selection.
 
         Args:
+            what: What to yank.
             sel: Use the primary selection instead of the clipboard.
-            title: Yank the title instead of the URL.
-            domain: Yank only the scheme, domain, and port number.
-            pretty: Yank the URL in pretty decoded form.
+            keep: If given, stay in visual mode after yanking.
         """
-        if title:
+        if what == 'title':
             s = self._tabbed_browser.page_title(self._current_index())
-            what = 'title'
-        elif domain:
+        elif what == 'domain':
             port = self._current_url().port()
             s = '{}://{}{}'.format(self._current_url().scheme(),
                                    self._current_url().host(),
                                    ':' + str(port) if port > -1 else '')
-            what = 'domain'
-        else:
+        elif what in ['url', 'pretty-url']:
             flags = QUrl.RemovePassword
-            if not pretty:
+            if what == 'url': # Not pretty
                 flags |= QUrl.FullyEncoded
             s = self._current_url().toString(flags)
-            what = 'URL'
+            what = 'URL' # For printing
+        elif what == 'selection':
+            caret = self._current_widget().caret
+            s = caret.selection()
+            if not caret.has_selection() or len(s) == 0:
+                message.info(self._win_id, "Nothing to yank")
+                return
 
```

Comment:
Please add an `else:  # pragma: no cover` here, which does `raise ValueError("Invalid value {!r} for 'what'".format(what))` or so. This should never happen, but if it does, it's better to do something defined.